Source/Core/DolphinWX/RmObjAddEdit.cpp

```diff
@@ -0,0 +1,338 @@
+	else if (RmObjEngine::GetRmObjTypeCharLength(tempEntries.at(0).type) <= 32)
+		EditRmObjValue->SetValue(wxString::Format("%0*X%08X%08X%08X", RmObjEngine::GetRmObjTypeCharLength(tempEntries.at(0).type)-24, (u32)((tempEntries.at(0).value_upper & 0xffffffff00000000) >> 32), (u32)(tempEntries.at(0).value_upper & 0xffffffff), (u32)((tempEntries.at(0).value_lower & 0xffffffff00000000) >> 32), (u32)((tempEntries.at(0).value_lower & 0xffffffff))));
+
+	wxButton* BruteForceUp = new wxButton(this, ID_BUTTON_UP, _("Up"));
+	wxButton* BruteForceDown = new wxButton(this, ID_BUTTON_DOWN, _("Down"));
+	//BruteForceUp->SetToolTip(_(""));
+	//BruteForceDown->SetToolTip(_(""));
+	BruteForceUp->Bind(wxEVT_BUTTON, &CRmObjAddEdit::ButtonUporDown, this);
+	BruteForceDown->Bind(wxEVT_BUTTON, &CRmObjAddEdit::ButtonUporDown, this);
+
+	wxBoxSizer* sEditRmObjName = new wxBoxSizer(wxHORIZONTAL);
+	sEditRmObjName->Add(EditRmObjNameText, 0, wxALIGN_CENTER_VERTICAL|wxALL, 5);
+	sEditRmObjName->Add(EditRmObjName, 1, wxEXPAND|wxALL, 5);
+	sEditRmObj->Add(sEditRmObjName, 0, wxEXPAND);
+	sbEntry = new wxStaticBoxSizer(wxVERTICAL, this, wxString::Format(_("Object Removal Code")));
+
+	wxGridBagSizer* sgEntry = new wxGridBagSizer(0, 0);
+	sgEntry->Add(EditRmObjType, wxGBPosition(0, 0), wxGBSpan(2, 5), wxEXPAND|wxALL, 5);
+	sgEntry->Add(EditRmObjValueText, wxGBPosition(2, 0), wxGBSpan(1, 1), wxALIGN_CENTER_VERTICAL|wxALL, 5);
+	sgEntry->Add(EditRmObjValue, wxGBPosition(2, 1), wxGBSpan(1, 1), wxEXPAND|wxALL, 5);
+	sgEntry->Add(BruteForceUp, wxGBPosition(2, 2), wxGBSpan(1, 1), wxEXPAND|wxALL, 5);
+	sgEntry->Add(BruteForceDown, wxGBPosition(2, 3), wxGBSpan(1, 1), wxEXPAND | wxALL, 5);
+	sgEntry->AddGrowableCol(1);
+
+	sbEntry->Add(sgEntry, 0, wxEXPAND);
+
+	sEditRmObj->Add(sbEntry, 0, wxEXPAND|wxALL, 5);
+	sEditRmObj->Add(CreateButtonSizer(wxOK | wxCANCEL), 0, wxEXPAND|wxLEFT|wxRIGHT|wxBOTTOM, 5);
+	SetSizerAndFit(sEditRmObj);
+	SetFocus();
+}
+
+void CRmObjAddEdit::SaveRmObjData(wxCommandEvent& event)
+{
+	if (!UpdateTempEntryData(itCurEntry))
+		return;
+
+	if (selection == -1)
+	{
+		for (int i = 0; i < rmObjCodes.size(); ++i)
+		{
+			if (rmObjCodes.at(i).name == WxStrToStr(EditRmObjName->GetValue()))
+			{
+				wxMessageBox(_("Name is already in use.  Please choose a unique name."), _("Error"));
+				return;
+			}
+		}
+
+		RmObjEngine::RmObj newRmObj;
+		newRmObj.name = WxStrToStr(EditRmObjName->GetValue());
+		newRmObj.entries = tempEntries;
+		newRmObj.active = true;
+
+		rmObjCodes.push_back(newRmObj);
+	}
+	else
+	{
+		rmObjCodes.at(selection).name = WxStrToStr(EditRmObjName->GetValue());
+		rmObjCodes.at(selection).entries = tempEntries;
+	}
+
+	AcceptAndClose();
+	event.Skip();
+}
+
+void CRmObjAddEdit::ButtonUporDown(wxCommandEvent& event)
+{
+	if (!UpdateTempEntryData(itCurEntry))
+		return;
+
```

Comment:
You may want to look into the use of validators, they would make all of this limit testing much smaller. `wxNumValidator` or a subclass of it could be handy here.